nautobot/docs/core-functionality/device-types.md

```diff
@@ -31,10 +31,12 @@ Whenever a new device is created, its components are automatically created per t
 * 48 templates for 1GE interfaces ("ge-0/0/0" through "ge-0/0/47")
 * Four templates for 10GE interfaces ("xe-0/2/0" through "xe-0/2/3")
 
-Once component templates have been created, every new device that you create as an instance of this type will automatically be assigned each of the components listed above.
+Once component templates have been created, every new device that you create as an instance of this type will automatically be assigned each of the components listed above. However, this automation occurs only when a device is first created: Adding or removing components on a device type does not automatically change existing device instances based on that device type.
```

Comment:
```suggestion
Once component templates have been created, every new device that you create as an instance of this type will automatically be assigned each of the components listed above. However, this automation occurs only when a device is first created: Adding or removing components on a device type does not automatically change existing device instances based on that device type.

Consider also that automatically changing components of existing devices when a device-type changes would require Nautobot to make very dangerous assumptions. For example, suppose you had a device type which included a specific line card with a specific set of interfaces. After creating some devices, representing your existing deployment of this device type, your deployment for new devices of this type changed to include a different line card and hence different interfaces, so you decided to update the device-type templates accordingly. If Nautobot were to automatically propagate this change to all existing devices of this type, it would result in an incorrect reflection of the reality that all existing devices have not yet been retrofitted with the new line card.
```